check.go

```diff
@@ -70,3 +111,228 @@ func checkMain(_ *cobra.Command, args []string) error {
 
 	return nil
 }
+
+func getExcludedLicenseTypes() []licenses.Type {
+	excludedLicenseTypes := make([]licenses.Type, 0)
+
+	if excludeForbidden {
+		excludedLicenseTypes = append(excludedLicenseTypes, licenses.Forbidden)
+	}
+
+	if excludeNotice {
+		excludedLicenseTypes = append(excludedLicenseTypes, licenses.Notice)
+	}
+
+	if excludePermissive {
+		excludedLicenseTypes = append(excludedLicenseTypes, licenses.Permissive)
+	}
+
+	if excludeReciprocal {
+		excludedLicenseTypes = append(excludedLicenseTypes, licenses.Reciprocal)
+	}
+
+	if excludeRestricted {
+		excludedLicenseTypes = append(excludedLicenseTypes, licenses.Restricted)
+	}
+
+	if excludeUnencumbered {
+		excludedLicenseTypes = append(excludedLicenseTypes, licenses.Unencumbered)
+	}
+
+	return excludedLicenseTypes
+}
+
+func isExcludedLicenseType(licenseType licenses.Type, excludedLicenseTypes []licenses.Type) bool {
+	for _, excluded := range excludedLicenseTypes {
+		if excluded == licenseType {
+			return true
+		}
+	}
+
+	return false
+}
+
+func getAllowedLicenseNames() []string {
+	if len(allowedLicenseNames) == 0 {
+		return []string{}
+	}
+
+	var allowed []string
+	supported := getSupportedLicenseNames()
+
+	for _, licenseName := range allowedLicenseNames {
+		if isAllowedLicenseName(licenseName, supported) {
+			allowed = append(allowed, licenseName)
+
+			continue
+		}
+
+		fmt.Fprintf(os.Stderr, "Unsupported license name %s provided, it will be ignored\n", licenseName)
+	}
+
+	return allowed
+}
+
+func isAllowedLicenseName(licenseName string, allowedLicenseNames []string) bool {
+	for _, allowed := range allowedLicenseNames {
+		if allowed == licenseName {
+			return true
+		}
+	}
+
+	return false
+}
+
+func getSupportedLicenseNames() []string {
```

Comment:
Curious, why do we need this list?

Reply:
I didn't find a method in github.com/google/licenseclassifier to get all defined license names.
It is used here to validate the input of flag `allowed_license_names`.

Reply:
Is there a strong need to do this validation? If there is a typo, the check command fails and reports the correct license name in error message.

An extraneous invalid license name seems not much harm either.